From df96dddc80c06472bbf83a388bc2312ee564285f Mon Sep 17 00:00:00 2001
From: Alan Winkowski <walan@marvell.com>
Date: Wed, 5 Dec 2018 19:10:28 +0200
Subject: [PATCH 0826/1051] net: mvpp2: fix when setting C2 to functional mode

Setting C2 to functional mode should be done once per
network controller, and not per port.

Change-Id: Ieb0f7e43073f73d88b15ee6058d833997d8bb438
Signed-off-by: Alan Winkowski <walan@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61552
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c
index b1a0612f639b..30af059768ef 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_cls.c
@@ -871,10 +871,6 @@ static void mvpp2_port_c2_cls_init(struct mvpp2_port *port)
 	c2.attr[0] = MVPP22_CLS_C2_ATTR0_QHIGH(qh) |
 		      MVPP22_CLS_C2_ATTR0_QLOW(ql);
 
-	/* Toggle C2 from Built-In Self-Test mode to Functional mode */
-	mvpp2_write(port->priv, MVPP2_CLS2_TCAM_CTRL_REG,
-		    MVPP2_CLS2_TCAM_CTRL_BYPASS_FIFO_STAGES);
-
 	mvpp2_cls_c2_write(port->priv, &c2);
 }
 
@@ -882,6 +878,10 @@ static void mvpp2_cls_c2_init(struct mvpp2 *priv)
 {
 	int index;
 
+	/* Toggle C2 from Built-In Self-Test mode to Functional mode */
+	mvpp2_write(priv, MVPP2_CLS2_TCAM_CTRL_REG,
+		    MVPP2_CLS2_TCAM_CTRL_BYPASS_FIFO_STAGES);
+
 	/* Invalidate all C2 entries */
 	for (index = 0; index < MVPP22_CLS_C2_MAX_ENTRIES; index++)
 		mvpp2_cls_c2_inv_set(priv, index);
-- 
2.17.1

